src/deadline/job_attachments/api/attachment.py

```diff
@@ -28,56 +33,156 @@ def attachment_download(
     If path mapping rules file is given, map to corresponding destinations.
 
     Args:
-        manifests: File Path to the manifest file for upload.
-        s3_root_path: S3 root path including bucket name and root prefix.
-        boto_session: Boto3 session for interacting with customer s3.
-        path_mapping_rules: Optional file path to a list of path mapping.
+        manifests (List[str]): File Path to the manifest file for upload.
+        s3_root_uri (str): S3 root uri including bucket name and root prefix.
+        boto3_session (boto3.Session): Boto3 session for interacting with customer s3.
+        path_mapping_rules (Optional[str], optional): Optional file path to a JSON file contains list of path mapping. Defaults to None.
+        logger (ClickLogger, optional): Logger to provide visibility. Defaults to ClickLogger(False).
     """
 
-    # path validation
-    if not all([os.path.isfile(manifest) for manifest in manifests]):
-        raise NonValidInputError(f"Specified manifests {manifests} contain invalid file.")
+    file_name_manifest_dict: Dict[str, BaseAssetManifest] = _get_file_name_manifest_dict(
+        manifests=manifests
+    )
+    path_mapping_rule_list: List[PathMappingRule] = _process_path_mapping(
+        path_mapping_rules=path_mapping_rules
+    )
+
+    merged_manifests_by_root: Dict[str, BaseAssetManifest] = dict()
+    for file_name in file_name_manifest_dict:
+        manifest: BaseAssetManifest = file_name_manifest_dict[file_name]
+        # File name is supposed to be prefixed by a hash of source path in path mapping, use that to determine destination
+        # If it doesn't appear in path mapping or mapping doesn't exist, download to current directory instead
+        destination = next(
+            (
+                rule.destination_path
+                for rule in path_mapping_rule_list
+                if rule.get_hashed_source_path(manifest.get_default_hash_alg()) in file_name
+            ),
+            # Write to current directory partitioned by manifest name when no path mapping defined
+            f"{os.getcwd()}/{file_name}",
+        )
+        # Assuming the manifest is already aggregated and correspond to a single destination
+        merged_manifests_by_root[destination] = manifest
```

Comment:
Should we have error handling in case there is a collision? Even though the current assumption is that it is not possible, might be good idea to code defensively here.